user/contactmethod/store.go

```diff
@@ -466,3 +477,39 @@ func (db *DB) FindAll(ctx context.Context, userID string) ([]ContactMethod, erro
 
 	return scanAll(rows)
 }
+
+func (db *DB) FindLastStatus(ctx context.Context, cmID string) (*n.MessageStatus, error) {
+	err := validate.UUID("Contact Method ID", cmID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = permission.LimitCheckAny(ctx, permission.User)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := db.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	var s n.MessageStatus
+	var lastStatus string
+	var hasNextRetry bool
+	row := wrapTx(ctx, tx, db.findLastStatus).QueryRowContext(ctx, cmID)
```

Comment:
`wrapTx` is a helper for when a func can take an optional transaction (e.g. `tx` could be nil). As written this code will always have a `tx` so `tx.StmtContext(db.findLastStatus)` would be more concise.